pkg/repository/v1_repository.go

```diff
@@ -153,16 +156,26 @@ func (r *V1Repository) UpdateComponents(specs []ComponentSpec) error {
 			}
 		}
 
-		reader, err := r.FetchComponent(versionItem)
+		if spec.Version == "" {
+			spec.Version = version
+		}
+
+		targetDir := filepath.Join(r.local.TargetRootDir(), localdata.ComponentParentDir, spec.ID, spec.Version)
+		target := filepath.Join(targetDir, versionItem.URL)
+		err = r.DownloadComponent(versionItem, target)
```

Comment:
I think we should delete the target file when the component installed successfully.

Reply:
Thanks for the advice, rm the target by default, and add an env named `TIUP_KEEP_SOURCE_TARGET` to keep the source target(maybe used for debugging propose).